quickwit-index-config/src/lib.rs

```diff
@@ -35,6 +35,7 @@ pub use default_index_config::{
     DefaultIndexConfig, DefaultIndexConfigBuilder, DocParsingError, FieldMappingEntry,
 };
 pub use error::QueryParserError;
+pub use query_builder::extract_tags_from_query;
```

Comment:
There is no great place to put the tag pruning logic because it is required by both search and indexing.
However index-config is probably the worst place :)

If possible `quickwit-common` would be ok but I think it requires adding a deps to tantivy_query_grammar which would be lame.

Maybe `quickwit-indexing` might be nicer? That will add a deps from search to `quickwit-indexing` though.
An alternative could be to create a dedicated quickwit-query project to avoid that dependency. No need to do that in this PR though.